apps/biobin/src/biobin.sh

```diff
@@ -0,0 +1,147 @@
+#!/bin/bash
+set -e -x -o pipefail
+
+main() {
+
+  ulimit -c unlimited
+  # switch to a temp directory and download all user input files
+
+  NUM_CORES=$(nproc)
+  TMPDIR=$(mktemp -d)
+  cd "$TMPDIR"
+  mkdir input
+  BIOBIN_ROLE_ARG=""
+  if [[ -n "$role" ]]; then
+    dx download "$role_rol" -o input/input.role
+    BIOBIN_ROLE_ARG="--role-file input/input.role"
+  fi
+  BIOBIN_PHENO_ARG=""
+  BIOBIN_TEST_ARG=""
+  if [[ -n "$phenotype_phe" ]]; then
+    dx download "$phenotype_phe" -o input/input.phenotype
+    BIOBIN_PHENO_ARG="--phenotype-file input/input.phenotype"
+    if [[ ${#regression_type[*]} -gt 0 ]]; then
+      BIOBIN_TEST_ARG="--test $(IFS="," ; echo "${regression_type[*]}")"
+    fi
+  fi
+  BIOBIN_COVAR_ARG=""
+  if [[ -n "$covariate_cov" ]]; then
+    dx download "$covariate_cov" -o input/input.covariate
+    BIOBIN_COVAR_ARG="--covariates input/input.covariate"
+  fi
+  BIOBIN_WEIGHT_ARG=""
+  if [[ -n "$weight_wgt" ]]; then
+    dx download "$weight_wgt" -o input/input.weight
+    BIOBIN_WEIGHT_ARG="--weight-file input/input.weight"
+  fi
+  BIOBIN_REGION_ARG=""
+  if [[ -n "$region_rgn" ]]; then
+    dx download "$region_rgn" -o input/input.region
+    BIOBIN_REGION_ARG="--region-file input/input.region"
+  fi
+  BIOBIN_INCLUDE_REGION_ARG=""
+  if [[ -n "$include_region_file" ]]; then
+    dx download "$include_region_file" -o input/input.gene
+    BIOBIN_INCLUDE_REGION_ARG="--include-region-file input/input.gene"
+  fi
+  VCF_FILE="input/input.vcf.gz"
+  TBI_FILE="$VCF_FILE.tbi"
+  dx download "$vcf_file" -o "$VCF_FILE"
+  tabix -p vcf "$VCF_FILE" 2>&1 | tee -a output.log
+
+  # fetch the executable(s) and shared resource file(s)
+  mkdir -p shared
+
+  if [ "${loki_db}" != "" ]; then
```

Comment:
loki_db is required, so it should be an error here if undefined

Reply:
The job wouldn't launch without the required inputs are specified. 
I removed the if condition checking as it becomes redundant.